docs/code/Flakes/Tutorials/Hacking.md

```diff
@@ -0,0 +1,32 @@
+---
+title: 03-Create a Development Environment
+date: 2021-02-09
+---
+
+## Hacking
+
+`nix-shell` has been replaced by `nix develop`, and by default it builds the attribute `devShell.x86_64-linux`, so a simple python development environment would look like this:
+
+`flake.nix`
+```
+{
+  description = "A very basic flake";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-21.11";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      pkgs = nixpkgs.legacyPackages.x86_64-linux
+    in {
+      devShell.x86_64-linux = pkgs.mkShell {
+        nativeBuildInputs = [
+          pkgs.python3
+        ];
+      };
+    };
+}
+```
```

Comment:
Can we cut down the examples so that they are still self-contained, but only deal with the topic of this page?
For example, remove pinning of nixpkgs, unless the page is about version pinning.

But here is a more interesting example of what could be put in the shell:
````suggestion
{
  description = "Flake with a development environment";

  outputs = { self, nixpkgs }: {
    devShell.x86_64-linux =
      with nixpkgs.legacyPackages.x86_64-linux;
      mkShell {
        nativeBuildInputs = [
          (python3.withPackages (ps: [ ps.numpy ps.toolz ps.requests ]))
        ];
      };
  };
}
```

This is an example of adding Python and some Python Wheels to the shell. More information (for those interested) can be found in the
[Nixpkgs Manual Python section](https://nixos.org/manual/nixpkgs/stable/#python).
````

Reply:
This builds on tutorial 2, and so I left the version pinning in to minimize required code changes from the last `flake.nix`, but feel free to remove.